controller/src/beerocks/cli/beerocks_cli_bml.cpp

```diff
@@ -1307,6 +1320,77 @@ int cli_bml::get_dcs_scan_results_caller(int numOfArgs)
     return -1;
 }
 
+/**
+ * Caller function for client_get_client_list_caller.
+ *
+ * @param [in] numOfArgs Number of received arguments
+ * @return 0 on success.
+ */
+int cli_bml::client_get_client_list_caller(int numOfArgs)
+{
+    if (numOfArgs == 0) {
+        return client_get_client_list();
+    }
+    return -1;
+}
+
+/**
+ * Caller function for client_set_client_caller.
+ *
+ * @param [in] numOfArgs Number of received arguments
+ * @return 0 on success.
+ */
+int cli_bml::client_set_client_caller(int numOfArgs)
+{
+
+    /*
+     * Mandatory:
+     *  sta_mac=<sta_mac>
+     * Optional:
+     *  selected_bands=<selected_bands>
+     *  stay_on_initial_radio=<stay_on_initial_radio>
+     *  stay_on_selected_device=<stay_on_selected_device>
+    ]*/
+    std::string::size_type pos;
+    std::string sta_mac(network_utils::WILD_MAC_STRING);
+    int8_t selected_bands          = BML_PARAMETER_NOT_CONFIGURED;
+    int8_t stay_on_initial_radio   = BML_PARAMETER_NOT_CONFIGURED;
+    int8_t stay_on_selected_device = BML_PARAMETER_NOT_CONFIGURED;
+    if (numOfArgs > 1) {
+        sta_mac = args.stringArgs[0];
+        for (int i = 1; i < numOfArgs; i++) { //first optional arg
+            if ((pos = args.stringArgs[i].find("selected_bands=")) != std::string::npos) {
+                selected_bands = beerocks::string_utils::stoi(
+                    args.stringArgs[i].substr(pos + sizeof("selected_bands")));
+            } else if ((pos = args.stringArgs[i].find("stay_on_initial_radio=")) !=
+                       std::string::npos) {
+                stay_on_initial_radio = string_utils::stoi(
+                    args.stringArgs[i].substr(pos + sizeof("stay_on_initial_radio")));
+            } else if ((pos = args.stringArgs[i].find("stay_on_selected_device=")) !=
+                       std::string::npos) {
+                stay_on_selected_device = string_utils::stoi(
+                    args.stringArgs[i].substr(pos + sizeof("stay_on_selected_device")));
+            }
+        }
+        client_set_client(sta_mac, selected_bands, stay_on_initial_radio, stay_on_selected_device);
+    }
+    return -1;
```

Comment:
```suggestion
    if (numOfArgs <= 1) {
        return -1;
    }
    
    sta_mac = args.stringArgs[0];
    for (int i = 1; i < numOfArgs; i++) { //first optional arg
        if ((pos = args.stringArgs[i].find("selected_bands=")) != std::string::npos) {
            selected_bands = beerocks::string_utils::stoi(
                args.stringArgs[i].substr(pos + sizeof("selected_bands")));
        } else if ((pos = args.stringArgs[i].find("stay_on_initial_radio=")) !=
                   std::string::npos) {
            stay_on_initial_radio = string_utils::stoi(
                args.stringArgs[i].substr(pos + sizeof("stay_on_initial_radio")));
        } else if ((pos = args.stringArgs[i].find("stay_on_selected_device=")) !=
                   std::string::npos) {
            stay_on_selected_device = string_utils::stoi(
                args.stringArgs[i].substr(pos + sizeof("stay_on_selected_device")));
        }
    }
    
    return client_set_client(sta_mac, selected_bands, stay_on_initial_radio, stay_on_selected_device);
```

note that in previous implementation you would always `return -1;`

Reply:
If you look at other `_caller` functions, they all follow this structure:
```
if (numOfArgs checks) {
     /*run command*/
}

return -1
```

But fixed the API call, so now the result of `client_set_client` will be returned